File: TaskManagementSystem.Domain/Models/Task.cs
using TaskManagementSystem.Domain.Enums;

namespace TaskManagementSystem.Domain.Models
{
    public class Task
    {
        public Guid Id { get; set; }
        public required string Title { get; set; }
        public string? Description { get; set; }
        public EPriority Priority { get; set; }
        public DateTime DueDate { get; set; }
        public ETaskStatus Status { get; set; }
    }
}

File: client-app/src/app/models/task.ts
import { Priority } from "../enums/Priority";
import { TaskStatus } from "../enums/TaskStatus";

export interface Task {
    id: string;
    title: string;
    description?: string;
    dueDate: Date | null;
    priority: Priority;
    status: TaskStatus;
}

export class Task implements Task {
    constructor(init?: TaskFormValues){
        Object.assign(this, init);
    }
}

export class TaskFormValues {
    id?: string = undefined;
    title: string = '';
    description?: string = '';
    dueDate: Date | null = null;
    priority: Priority = Priority.Low;
    status: TaskStatus = TaskStatus.Pending;

    constructor(task?: Task){
        if(task){
            this.id = task.id;
            this.title = task.title;
            this.description = task.description;
            this.dueDate = task.dueDate;
            this.priority = task.priority;
            this.status = task.status;
        }
    }
}